### DISCUSSION TITLE
Can I send metrics to existing web server, instead of starting separate server (start_http_server)

### DISCUSSION THREAD

**Discussion by 29swastik** [prometheus/client_python#884] [Announcements]
I've a python file running KafkaConsumer in infinite loop, there I want to increment prometheus counter whenever there is a failure while processing the consumed kafka message. 

Here I don't want run `start_http_server` because I already have one fastapi server running and I want to send my counter metric to that fastapi server instead of running separate `start_http_server`, how can I do that?

**Comment by mcastagnetti** [comment]
It seems that some prometheus exporters exists for Starlette based apps (such as FastAPI) like https://github.com/stephenhillier/starlette_exporter

But I'm also wondering if the official prometheus python client could provide an exporter instead of starting a distinct http server 🙏

**Comment by csmarchbanks** [comment]
This can be accomplished in a handful of ways. Internally we also use `handle_metrics` form starlette_exporter as mentioned in the other comment. In addition, you can use the `make_asgi_app` function as described at https://github.com/prometheus/client_python#asgi, or even manually call `generate_latest` in your own handler. All of these options allow you to avoid starting a separate http server.
